### DISCUSSION TITLE
How to increase severity for `argument name not accessed`

### DISCUSSION THREAD

**Discussion by jugmac00** [microsoft/pylance-release#3018] [Q&A] [answered]
Hey,

I had code like this...

```python
def build_secrets(distribution_name: str) -> dict:
    try:
        credentials = config["artifactory"]["read_credentials"]
    except NoSectionError:
        return {}
    return {"read_auth": credentials}
```

While `distribution_name` has a slightly different color and when you hover over it Pylance shows the mentioned warning, there is no wiggly line, no entry in the "PROBLEMS" tab... which meant I did not get warned by VS Code and proposed this code - luckily my error was caught in a review.

How would I turn this warning into an error?

I have already scrolled though pylance's `package.json` but I did not find a corresponding setting.

Thank you!

**Comment by erictraut** [comment, marked as answer]
Pylance does not emit diagnostics (errors, warnings, etc.) for an unused parameter, so there's no way to adjust the severity. There are many legitimate cases in Python for function parameters that must be present but go unused (e.g. magic methods, abstract method implementations, protocol method implementations, base class method overrides), so emitting a diagnostic for this case would be produce many false positives. Instead, Pylance uses a subtle "grayed out" effect so you can see that a parameter is unreferenced and decide whether that's appropriate in your particular case.

**Reply by jugmac00** [reply]
Eric,
Thanks for your quick reply.

The color difference for unused parameters is indeed subtle - I did not spot it, especially as it was the only parameter in that function. I will try a different theme.

Thanks for giving an explanation why it would make no sense to report all unused parameters.

While `pylint` would spot this error, I prefer not to use it for the exact reason you mentioned in your answer, it reports many false-positives.
